plugins/pulp_rpm/plugins/importers/yum/existing.py

```diff
@@ -129,6 +129,20 @@ def check_all_and_associate(wanted, conduit, download_deferred):
     return still_wanted
 
 
+def update_unit_downloaded_flag(unit):
+    """
+    Updates unit's download flag to True once it was placed to its storage location.
+
+    :param unit: A content unit.
+    :type unit: pulp_rpm.plugins.db.models.RPM/DRPM/Distribution
+    """
+
+    unit_model = plugin_api.get_unit_model_by_id(unit._content_type_id)
+    if unit._storage_path and os.path.exists(unit._storage_path):
+        unit_qs = unit_model.objects(_storage_path=unit._storage_path)
+        unit_qs.update_one(set__downloaded=True)
```

Comment:
Why do this search based on storage path, and not just update the unit by it's id?